Bottom: 6ce34ecf4f21090f527775aca9b4b13f06fb26c7
Top:    99cba68d5eef592e28eca3853ea9912ff5e2d3cc
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2013-10-27 16:57:15 +0000

Hack to avoid GL calls when we're out of context


---

diff --git a/src/hid/gtk/gtkhid-gl.c b/src/hid/gtk/gtkhid-gl.c
index dc2f783..e27f08c 100644
--- a/src/hid/gtk/gtkhid-gl.c
+++ b/src/hid/gtk/gtkhid-gl.c
@@ -486,6 +486,15 @@ set_gl_color_for_gc (hidGC gc)
     return;
 
   free (priv->current_colorname);
+  priv->current_colorname = NULL;
+
+  /* If we can't set the GL colour right now, quit with
+   * current_colorname set to NULL, so we don't NOOP the
+   * next set_gl_color_for_gc call.
+   */
+  if (!priv->in_context)
+    return;
+
   priv->current_colorname = strdup (gc->colorname);
   priv->current_alpha_mult = gc->alpha_mult;
 
@@ -549,9 +558,6 @@ set_gl_color_for_gc (hidGC gc)
 #endif
   }
 
-  if(!priv->in_context)
-    return;
-
   hidgl_flush_triangles (&buffer);
   glColor4d (r, g, b, a);
 }
@@ -834,6 +840,9 @@ draw_crosshair (render_priv *priv)
   static int done_once = 0;
   static GdkColor cross_color;
 
+  if (!priv->in_context)
+    return;
+
   if (!done_once)
     {
       done_once = 1;
